scene/gui/graph_node.cpp

```diff
@@ -524,13 +554,43 @@ bool GraphNode::is_slot_enabled_right(int p_idx) const {
 	return slot_info[p_idx].enable_right;
 }
 
+void GraphNode::set_slot_enabled_right(int p_idx, bool p_enable_right) {
+	ERR_FAIL_COND_MSG(p_idx < 0, vformat("Cannot set enable_right because Slot ID p_idx = %d is smaller than 0.", p_idx));
+
+	slot_info[p_idx].enable_right = p_enable_right;
+	update();
+	connpos_dirty = true;
+
+	emit_signal("slot_updated", p_idx);
+}
+
+void GraphNode::set_slot_type_right(int p_idx, int p_type_right) {
+	ERR_FAIL_COND_MSG(!slot_info.has(p_idx), vformat("Cannot set type_right beacuse slot %d doesn't exist.", p_idx));
+
+	slot_info[p_idx].type_right = p_type_right;
+	update();
+	connpos_dirty = true;
+
+	emit_signal("slot_updated", p_idx);
+}
+
 int GraphNode::get_slot_type_right(int p_idx) const {
 	if (!slot_info.has(p_idx)) {
 		return 0;
 	}
 	return slot_info[p_idx].type_right;
 }
 
+void GraphNode::set_slot_color_right(int p_idx, const Color &p_color_right) {
+	ERR_FAIL_COND_MSG(!slot_info.has(p_idx), vformat("Cannot set color_right beacuse slot %d doesn't exist.", p_idx));
```

Comment:
```suggestion
	ERR_FAIL_COND_MSG(!slot_info.has(p_idx), vformat("Cannot set color_right for a slot '%d' because it hasn't been enabled.", p_idx));
```